compiler/src/iree/compiler/Codegen/SPIRV/KernelConfig.cpp

```diff
@@ -235,6 +243,75 @@ LogicalResult setConvOpConfig(linalg::LinalgOp linalgOp,
 
 }  // namespace detail
 
+bool hasTwoOrThreeLoopsInfo(linalg::LinalgOp linalgOp) {
+  return linalgOp.getNumParallelLoops() >= 2 &&
+         linalgOp.getNumParallelLoops() <= 3;
+}
+
+namespace detail {
+
+LogicalResult setTransposeConfig(linalg::LinalgOp linalgOp) {
+  LinalgOpInfo opInfo(linalgOp, sharedMemTransposeFilter);
+
+  // Checks preconditions for shared mem transpose.
+  if (!opInfo.isTranspose() || opInfo.isDynamic() || opInfo.isReduction() ||
```

Comment:
I hope this isnt for just a transpose. If so, this seems too specific a thing to set configuration for. This should be folded into generic op lowering configuration. I raised this issue when this was done on CUDA side, and I was told that this was the plan, but now it is moving to other backends as well. I am not sure this is a good development.